ww3tools/ProcSat_Altimeter.py

```diff
@@ -0,0 +1,296 @@
+              
+            # Coordinates of a point
+            firstcoord = (lat[0], lon[0])
+            segtrack = list(zip(lat, lon))
+            # Calculate distances to reference coordinates
+            distances = np.array([geodesic(firstcoord, ref_coords).meters for ref_coords in segtrack])
+            # plot(distances,'.')
+            segments = np.array(distances // wconfig['dlim']).astype('int')
+            # print(' Total of '+repr(len(np.unique(segments)))+' segments')
+            # 1h (+-30min) JASON3 2.850 Kms (487 records). Total of 82 segments of 35km
+            for j in range(0,len(np.unique(segments))):
+                indseg=np.where(segments==int(j))
+                if np.size(indseg)>=1:
+                    indseg=indseg[0]
+
+                    stime = np.append(stime,atime[t])
+                    rstime = np.append(rstime,np.nanmean(AODN['TIME'][indt[indseg]]))
+                    slat = np.append(slat,np.nanmean(AODN['LATITUDE'][indt[indseg]]))
+                    slon = np.append(slon,np.array(AODN['LONGITUDE'][indt[indseg]])[int(np.round(len(indseg)/2))]) # avoid problems with 360-0
+                    swdepth = np.append(swdepth,np.nanmean(AODN['WDEPTH'][indt[indseg]]))
+                    swdistcoast = np.append(swdistcoast,np.nanmean(AODN['DISTCOAST'][indt[indseg]]))
+                    shs = np.append(shs,np.nanmean(AODN['HS'][indt[indseg]]))
+                    shscal = np.append(shscal,np.nanmean(AODN['HS_CAL'][indt[indseg]]))
+                    swsp = np.append(swsp,np.nanmean(AODN['WSPD'][indt[indseg]]))
+                    swspcal = np.append(swspcal,np.nanmean(AODN['WSPD_CAL'][indt[indseg]]))
+                    # print(repr(t)+" "+repr(i))
+                    i=i+1
+
+    if wconfig['qc']==0:
+        indq = np.where( (stime>=adatemin) & (stime<=adatemax) )
+    else:
+        indq = np.where( (swdepth>=wconfig['mindepth']) & (swdistcoast>=wconfig['mindfc']) &
+            (shs>wconfig['hsmin']) & (shs<wconfig['hsmax']) & (swsp>wconfig['wspmin']) & (swsp<wconfig['wspmax']) &
+            (stime>=adatemin) & (stime<=adatemax) )
+
+    if np.size(indq)>2:
+        indq=indq[0]
+        procaodn = {'TIME': stime[indq],'SAT_TIME': rstime[indq], 'LATITUDE': slat[indq], 'LONGITUDE': slon[indq],
+            'WDEPTH': swdepth[indq], 'DISTCOAST': swdistcoast[indq],
+            'HS': shs[indq], 'HS_CAL': shscal[indq],
+            'WSPD': swsp[indq], 'WSPD_CAL': swspcal[indq]}
+
+    else:
+        procaodn = {'TIME': [],'SAT_TIME': [], 'LATITUDE': [], 'LONGITUDE': [],
+            'WDEPTH': [], 'DISTCOAST': [],
+            'HS': [], 'HS_CAL': [],
+            'WSPD': [], 'WSPD_CAL': []}
+
+    AODN_ALONGTRACK = pd.DataFrame(procaodn)
+    return AODN_ALONGTRACK
+    del 
+
+
+# weight function for pyresample
+def wf(pdist):
+    a=(1 - pdist / (dlim+1))
+    return (abs(a)+a)/2
+
+# the old precessing, collocation to model grid, from gridSatGlobal_Altimeter, will enter here
```

Comment:
Is this comment old or is this a remaining to do?

Reply:
As soon as I include def gridded(AODN,atime,wconfig) I will delete this comment. 
This function will include the old gridSatGlobal_Altimeter.py method.

Reply:
Sounds good!